kuma/javascript/src/ssr.jsx

```diff
@@ -22,7 +24,53 @@ export default function ssr(data) {
             });return(v===true)?1:((v===false)?0:v);`
         );
     }
-    localize(data.requestData.locale, data.stringCatalog, pluralFunction);
+    localize(data.locale, data.stringCatalog, pluralFunction);
 
-    return renderToString(<App documentData={data.documentData} />);
+    let html = '';
+    switch (componentName) {
+        case 'document':
+            html = renderToString(
+                <DocumentPage documentData={data.documentData} />
+            );
+            break;
+        case 'landing':
+            html = renderToString(<LandingPage />);
+            break;
+        default:
+            console.error(
+                'Can not render unknown component name:',
+                componentName
+            );
+            break;
+    }
+
+    // TODO: the server-side rendered HTML will include a lot of
+    // separate little stylesheets becasue that is the way Emotion
+    // outputs them when server-side rendering. Arguably, this might be
+    // the fastest way to render content above-the-fold because styles
+    // further down in the document don't need to be parsed until they
+    // are actually needed. But it does seem like it would be inefficient
+    // to have dozens of separate stylesheets to parse. We need data
+    // on this, I suppose.
+    //
+    // I attempted to consolidate the stylesheets with the following
+    // code, but it turned out to break the hydration process because
+    // emotion expect to have all of the separate stylesheets. So if you
+    // uncomment this code, you'll get a single big stylesheet, but the
+    // page will re-render after loading, which is worse than the problem
+    // we were trying to solve:
+    //
+    // Post-process the HTML string we to consolidate
+    // all of the emotion stylesheets into a single one.
+    // let stylesheets = [];
+    // html = html.replace(
+    //     /<style data-emotion-css="[^"]+">(.*?)<\/style>/g,
+    //     function(match, styles) {
+    //         stylesheets.push(styles);
+    //         return '';
+    //     }
+    // );
+    // return `<style>\n${stylesheets.join('\n')}\n</style>\n${html}`;
+    //
```

Comment:
@peterbe: this long comment is here for your benefit. I tried to consolidate all of the stylesheets, and found that it broke the hydration process. And in the process I realized that maybe lots of little stylesheets is not that inefficient anyway.  

But anyway, this comment and the comment code are preserved here in case you want to attempt the consolidation again in the future.

Reply:
That should not be a blocker at this stage I don't think.